torch/_dynamo/output_graph.py

```diff
@@ -313,6 +318,7 @@ def __init__(
                 export=self.export,
             )
         self.tracing_context: TracingContext = TracingContext(fake_mode)
+        self.compile_context: Optional[CompileContext] = CompileContext.try_get()
```

Comment:
Saving the entire compile context seems sus. In particular, the compile context keeps track of what attempt you are for Dynamo retracing, this quantity is meaningless once you're doing export passes.

Reply:
The motivation of saving the entire CompileContext here was to avoid recreating it when passing it on to the decorator outside of dynamo, the attempt was the only other thing tracked here. But I can change this to save only CompileId here and we can recreate the context when passing it on the the decorator.

Reply:
Yes, recreating is more semantically correct